src/cli/commands/create.js

```diff
@@ -16,17 +16,50 @@ export function hasWrapper(commander: Object, args: Array<string>): boolean {
   return true;
 }
 
+const coerceName = (name = '') => (name === '' ? 'create' : `create-${name}`);
+const coerceScope = (scope = '') => (scope === '@' ? '' : scope);
+const coerceFullName = (scope, name) => [coerceScope(scope), coerceName(name)].filter(Boolean).join('/');
+
+/**
+ * # Tests
+ *
+ * ## basic
+ * parseBuilderName('name').packageName === 'create-name'
+ * parseBuilderName('@scope/name').packageName === '@scope/create-name'
+ *
+ * ## not adding "-" if name is empty
+ * parseBuilderName('@scope/').packageName === '@scope/create'
+ * parseBuilderName('@scope').packageName === '@scope/create'
+ *
+ * ## edge cases
+ * parseBuilderName('@/name').packageName === 'create-name'
+ * parseBuilderName('/name').packageName === 'create-name'
+ * parseBuilderName('@/').packageName === 'create'
+ */
+const parseBuilderName = str => {
+  const parts = str.split('/');
```

Comment:
It should reject as invalid if there's more than two parts, or if the first part doesn't start with a '@'

Reply:
good idea

Reply:
I changed `parseBuilderName` to `parsePackageName` and it throws error if first character is `.` or `/`. It also throws if scope equals `@`. 

About throwing id there are more than 2 parts: and i want yarn to be robust so I just put that extra part into `path` field of `pkgNameObject`¹ and ignore it in rest of `create` command.

\[1]:
```js
'@scope/name/path/to/file.js' => pkgNameObject {
  fullName: '@scope/name',
  name: 'name',
  scope: '@scope',
  path: 'path/to/file.js',
  full: '@scope/name/path/to/file.js',
}
```